packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -473,9 +473,22 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 					"Attempted to process an inbound signal without a handler attached",
 				);
 			}
+
+			const envelope = JSON.parse(message.content as string);
+			const targetClientId = message.targetClientId ?? envelope.targetClientId ?? undefined;
+
+			// Drop signal if target client ID it does not match the receiving client ID
+			if (
+				targetClientId !== undefined &&
+				targetClientId !== this.connectionManager.clientId
+			) {
+				return;
+			}
```

Comment:
Logic exists that Signals sent come back as a way of monitoring system health. Targeting means they won't come back - the `local` parameter when listening won't be true for any non-self-targeted signals. We need an update to that logic to support targeted signals.
Also, the Notifications feature of DIS will benefit from being able to see that a targeted signal sent without service support came back. Should be a consideration in the design/implementation with trade-offs noted.

Reply:
Definitely. To align with the signal metrics with track (round trip latency and dropped signals) I added a check to include tracking all non-targeted signals and self-targeted signals, since these are signals we expect to return to the sending client. 

I also removed the dropping of signals based on target client id, since this information could be beneficial to receiving clients in some cases. Clients can decide to ignore received signals that were not targeted to them.